tests/test.cpp

```diff
@@ -4324,6 +4324,78 @@ void FlatbuffersIteratorsTest() {
 void FlatbuffersIteratorsTest() {}
 #endif
 
+void PrivateAnnotationsLeaks() {
+  // Simple schemas and a "has optional scalar" sentinal.
+  std::vector<std::string> schemas;
+  schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; }");
+  schemas.push_back("table Monster { mana : int; } struct ABC { mana: int; }");
+
+  schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; } union Any (private) { Monster, ABC }");
+
+  schemas.push_back(
+      "table Monster { mana : int; } struct ABC { mana: int; } union Any { "
+      "Monster, ABC }");
+  schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; } enum Race:byte (private) { None = -1, Human = 0, }");
+  schemas.push_back(
+      "table Monster { mana : int; } struct ABC { mana: int; } enum Race:byte "
+      "{ None = -1, Human = 0, }");
+  schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; } enum Race:byte (private) { None = -1, Human = 0, } union Any "
+      "(private) { Monster, ABC }");
+  schemas.push_back(
+      "table Monster { mana : int; } struct ABC { mana: int; } enum Race:byte "
+      "{ None = -1, Human = 0, } union Any { Monster, ABC }");
+
+  std::vector<std::string> failure_schemas;
+  failure_schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC { mana: int; }");
+  failure_schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC { mana: int; } union "
+      "Any { Monster, ABC }");
+  failure_schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC { mana: int; } enum "
+      "Race:byte { None = -1, Human = 0, }");
+  failure_schemas.push_back(
+      "table Monster (private) { mana : int; } struct ABC (private) { mana: "
+      "int; } enum Race:byte { None = -1, Human = 0, } union Any { Monster, "
+      "ABC }");
+
+  flatbuffers::IDLOptions opts;
+  opts.lang_to_generate = flatbuffers::IDLOptions::Language::kSwift;
+  opts.no_leak_private_annotations = true;
+
+  for (auto schema = schemas.begin(); schema < schemas.end(); schema++) {
+    flatbuffers::Parser parser(opts);
+    TEST_ASSERT(parser.Parse(schema->c_str()));
+  }
+
+  for (auto schema = failure_schemas.begin(); schema < failure_schemas.end();
+       schema++) {
+    flatbuffers::Parser parser(opts);
+    TEST_EQ(false, parser.Parse(schema->c_str()));
+  }
+
+  opts.no_leak_private_annotations = false;
+
+  for (auto schema = schemas.begin(); schema < schemas.end(); schema++) {
+    flatbuffers::Parser parser(opts);
+    TEST_ASSERT(parser.Parse(schema->c_str()));
+  }
+
+  for (auto schema = failure_schemas.begin(); schema < failure_schemas.end();
+       schema++) {
+    flatbuffers::Parser parser(opts);
+    TEST_ASSERT(parser.Parse(schema->c_str()));
+  }
```

Comment:
Wrote cpp code, to verify that we don't break the current usage, and to verify that it actually works